pkg/hubble/parser/seven/http.go

```diff
@@ -80,3 +82,37 @@ func (p *Parser) httpSummary(flowType accesslog.FlowType, http *accesslog.LogRec
 	}
 	return ""
 }
+
+// filterHeader receives a key-value pair of an http header along with an HubbleRedactSettings.
+// Based on the allow/deny lists of the provided HttpHeadersList it returns the original value
+// or the redacted constant "HUBBLE_REDACTED" accordingly:
+//  1. If HubbleRedactSettings is enabled (meaning that hubble.redact feature is enabled) but both allow/deny lists are empty then the value of the
+//     header is redacted by default.
+//  2. If the header's key is contained in the allow list then the value
+//     of the header will not be redacted.
+//  3. If the header's key is contained in the deny list then the value
+//     of the header will be redacted.
+//  4. If none of the above happens, then if there is any allow list defined then the value will be redacted
+//     otherwise if there is a deny list defined the value will not be redacted.
+func filterHeader(key string, value string, redactSettings options.HubbleRedactSettings) string {
+	if redactSettings.Enabled {
+		if len(redactSettings.RedactHttpHeaders.Allow) == 0 && len(redactSettings.RedactHttpHeaders.Deny) == 0 {
+			// That's the default case, where redact is generally enabled but not headers' lists
+			// have been specified. In that case we redact everything by default.
+			return defaults.SensitiveValueRedacted
+		}
+
+		if _, ok := redactSettings.RedactHttpHeaders.Allow[strings.ToLower(key)]; ok {
+			return value
+		} else if _, ok = redactSettings.RedactHttpHeaders.Deny[strings.ToLower(key)]; ok {
+			return defaults.SensitiveValueRedacted
+		} else {
+			if len(redactSettings.RedactHttpHeaders.Allow) > 0 {
+				return defaults.SensitiveValueRedacted
+			} else if len(redactSettings.RedactHttpHeaders.Deny) > 0 {
+				return value
+			}
+		}
+	}
+	return value
```

Comment:
Generally speaking, returning early makes the code easier to read as it limits indentation and cognitive overhead. What do you think about the following changes for the function body?
```go
	if !redactSettings.Enabled {
		return value
	}
	if len(redactSettings.RedactHttpHeaders.Allow) == 0 && len(redactSettings.RedactHttpHeaders.Deny) == 0 {
		// That's the default case, where redact is generally enabled but not headers' lists
		// have been specified. In that case we redact everything by default.
		return defaults.SensitiveValueRedacted
	}

	if _, ok := redactSettings.RedactHttpHeaders.Allow[strings.ToLower(key)]; ok {
		return value
	}
	if _, ok = redactSettings.RedactHttpHeaders.Deny[strings.ToLower(key)]; ok {
		return defaults.SensitiveValueRedacted
	}

	if len(redactSettings.RedactHttpHeaders.Allow) > 0 {
		return defaults.SensitiveValueRedacted
	}
	return value
```

Reply:
Nice one! Looks more tidy like this, thanks!